src/transitions/cond/MCCondEnqueue.cpp

```diff
@@ -1,28 +1,105 @@
 #include "mcmini/transitions/cond/MCCondEnqueue.h"
 #include "mcmini/MCTransitionFactory.h"
-#include "mcmini/mcmini_private.h"
+#include "mcmini/mcmini_private.h" /* For trace_pid, and state */
 #include "mcmini/transitions/mutex/MCMutexTransition.h"
 #include "mcmini/transitions/mutex/MCMutexUnlock.h"
+#include "mcmini/misc/cond/MCConditionVariableArbitraryPolicy.hpp"
+#include <pthread.h>
+#include <string.h>
+#include <sys/uio.h> /* For process_vm_readv() */
 
 MCTransition *
 MCReadCondEnqueue(const MCSharedTransition *shmTransition,
                   void *shmData, MCStack *state)
 {
-  const auto shmCond =
-    static_cast<MCSharedMemoryConditionVariable *>(shmData);
-  const auto condInShm     = shmCond->cond;
-  const auto mutexInShm    = shmCond->mutex;
-  const auto condSystemId  = (MCSystemID)condInShm;
-  const auto mutexSystemId = (MCSystemID)mutexInShm;
-  const auto condThatExists =
-    state->getVisibleObjectWithSystemIdentity<MCConditionVariable>(
-      condSystemId);
+  // WAS: const MCSystemID condSystemId = *condInShm; condInShm=shmData;
+  pthread_cond_t * condSystemId = *static_cast<pthread_cond_t **>(shmData);
+  // WAS: const auto mutexSystemId = (MCSystemID)mutexInShm;
+  auto shmCond = static_cast<MCSharedMemoryConditionVariable *>(shmData);
+  const auto mutexSystemId = (MCSystemID)shmCond->mutex;
   const auto mutexThatExists =
     state->getVisibleObjectWithSystemIdentity<MCMutex>(mutexSystemId);
```

Comment:
We should handle the case with a mutex unknown to the model as well as we do in `MCMutexLock` now. The same code would apply here.

Reply:
@maxwellpirtle ,
    Could we discuss this issue?  I was thinking about it, and I then added the following FIXME for MCReadCondEnqueue

>   // FIXME:  We still need to verify that if the condition variable had
//         previously been associated with a certain mutex, then
//         the new mutex is the same as the old one.  Technically,
//         it's legal to do both:
//         &nbsp;&nbsp;  pthread_cond_wait(&cond, &mutex1);
//         &nbsp;&nbsp;  pthread_cond_wait(&cond, &mutex2);
//         When the thread waits, it must remember what mutex it will wait on.